bin/peggy-cli.mjs

```diff
@@ -0,0 +1,593 @@
+import { Command, CommanderError, InvalidArgumentError, Option } from "commander";
+import fs from "fs";
+import path from "path";
+import { default as peggy } from "../lib/peg.js";
+import util from "util";
+
+export { CommanderError, InvalidArgumentError };
+
+// Options that aren't for the API directly:
+const PROG_OPTIONS = ["input", "output", "sourceMap", "test", "testFile", "verbose"];
+const MODULE_FORMATS = ["amd", "bare", "commonjs", "es", "globals", "umd"];
+const MODULE_FORMATS_WITH_DEPS = ["amd", "commonjs", "es", "umd"];
+const MODULE_FORMATS_WITH_GLOBAL = ["globals", "umd"];
+
+// Helpers
+
+function select(obj, sel) {
+  const ret = {};
+  for (const s of sel) {
+    if (Object.prototype.hasOwnProperty.call(obj, s)) {
+      ret[s] = obj[s];
+      delete obj[s];
+    }
+  }
+  return ret;
+}
+
+function commaArg(val, prev) {
+  return (prev || []).concat(val.split(",").map(x => x.trim()));
+}
+
+// Files
+
+function readStream(inputStream) {
+  return new Promise((resolve, reject) => {
+    const input = [];
+    inputStream.on("data", data => { input.push(data); });
+    inputStream.on("end", () => resolve(Buffer.concat(input).toString()));
+    inputStream.on("error", er => {
+      Error.captureStackTrace(er);
+      reject(er);
+    });
+  });
+}
+
+function readFile(name) {
+  let f = null;
+  try {
+    f = fs.readFileSync(name, "utf8");
+  } catch (e) {
+    throw new InvalidArgumentError(`Can't read from file "${name}".`, e);
+  }
+  return f;
+}
+
+/**
+ * @typedef {object} Stdio
+ * @property {stream.Readable} [in] StdIn.
+ * @property {stream.Writable} [out] StdOut.
+ * @property {stream.Writable} [err] StdErr.
+ */
+
+// Command line processing
+export class PeggyCLI extends Command {
+  /**
+   * Create a CLI environment.
+   *
+   * @param {Stdio} [stdio] Replacement streams for stdio, for testing.
+   */
+  constructor(stdio) {
+    super("peggy");
+
+    /** @type {Stdio} */
+    this.std = Object.assign({
+      in: process.stdin,
+      out: process.stdout,
+      err: process.stderr,
+    }, stdio);
+
+    /** @type object */
+    this.argv = {};
```

Comment:
Types may be enclosed in braces for consistency. Also, maybe:
```suggestion
    /** @type {peggy.BuildOptionsBase} */
    this.argv = {};
```

Reply:
Done.